src/ngx_http_modsecurity_module.c

```diff
@@ -622,6 +625,20 @@ ngx_http_modsecurity_create_main_conf(ngx_conf_t *cf)
     msc_set_connector_info(conf->modsec, MODSECURITY_NGINX_WHOAMI);
     msc_set_log_cb(conf->modsec, ngx_http_modsecurity_log);
 
+    /* Set up audit log reload */
+    ngx_str_t log_reload_file = ngx_string("/dev/null");
+    conf->audit_log_reload = ngx_conf_open_file(cf->cycle, &log_reload_file);
+    if (conf->audit_log_reload == NULL) {
+        dd("failed to open file for triggering log reload");
+        return NGX_CONF_ERROR;
+    }
+    conf->audit_log_reload->data = ngx_list_create(cf->pool, 100, sizeof(Rules*));
```

Comment:
We don't currently free this list as part of a cleanup action. This may be fine if the memory is small enough to be allocated in the pool and automatically freed when the configuration is reloaded.

I need to understand this better.

Reply:
Phew. It seemed like that was correct, but I had a foggy memory of reading something about an nginx function that frees only memory allocated outside a pool. I haven't been able to find that statement again though.